crates/cojson-core-rn/pkg/android/CMakeLists.txt

```
@@ -0,0 +1,101 @@
+cmake_minimum_required(VERSION 3.10)
+project(CoJSONCoreRN)
+
+set(PACKAGE_NAME CoJSONCoreRN)
+set(CMAKE_VERBOSE_MAKEFILE ON)
+set(CMAKE_CXX_STANDARD 20)
+
+# Architecture mapping for Rust targets
+if(ANDROID_ABI STREQUAL "arm64-v8a")
+    set(RUST_TARGET_ARCH "arm64-v8a")
+elseif(ANDROID_ABI STREQUAL "armeabi-v7a")
+    set(RUST_TARGET_ARCH "armeabi-v7a")
+elseif(ANDROID_ABI STREQUAL "x86")
+    set(RUST_TARGET_ARCH "x86")
+elseif(ANDROID_ABI STREQUAL "x86_64")
+    set(RUST_TARGET_ARCH "x86_64")
+else()
+    message(FATAL_ERROR "Unsupported Android ABI: ${ANDROID_ABI}")
+endif()
+
+# Add a library from the source files
+add_library(${PACKAGE_NAME} SHARED
+  "src/main/cpp/cpp-adapter.cpp"
+  "../cpp/HybridCoJSONCoreRN.cpp"
+)
+
+# add Nitrogen specs
+include(${CMAKE_SOURCE_DIR}/../nitrogen/generated/android/CoJSONCoreRN+autolinking.cmake)
+
+# Specify include directories
+include_directories(
+  "src/main/cpp"
+  "../cpp"
+  "../build/includes"
+  "../build/includes/rust"
+  "../nitrogen/generated/shared/c++"
+)
+
+# Third party libraries (Prefabs)
+find_library(LOG_LIB log)
+
+# Map Android ABI to Rust target for both cxxbridge and static library paths
+if(ANDROID_ABI STREQUAL "arm64-v8a")
+    set(RUST_TARGET "aarch64-linux-android")
+elseif(ANDROID_ABI STREQUAL "armeabi-v7a")
+    set(RUST_TARGET "armv7-linux-androideabi")
+elseif(ANDROID_ABI STREQUAL "x86")
+    set(RUST_TARGET "i686-linux-android")
+elseif(ANDROID_ABI STREQUAL "x86_64")
+    set(RUST_TARGET "x86_64-linux-android")
+endif()
+
+# Import Rust static library (after RUST_TARGET is defined)
+set(CRATES_DIR "/Users/brad/dev/jazz/crates")
```

```suggestion
set(CRATES_DIR "${CMAKE_SOURCE_DIR}/../../../")
```

Started debugging the build issues for android, found still getting other errors.